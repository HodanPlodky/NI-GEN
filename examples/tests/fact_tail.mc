int fact_inner(int n, int acc) {
    if (n <= 1) 
        return acc;
    return fact_inner(n - 1, acc * n);
}

int fact(int n) {
    return fact_inner(n, 1);
}

int main() {
    return fact(5);
}
