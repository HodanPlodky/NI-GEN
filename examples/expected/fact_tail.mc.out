input file : examples/tests/fact_tail.mc
return 120
